### PR TITLE
Mutant colors is now stored in the UI dna data

### LINKED ISSUE #210: Giving nonhuman races the UI or UI+UE of another nonhuman of the same race does not change appearance
After printing out a lizardman UI+UE from the genetics console and using it on another lizardman, only the name and eye color of the second lizard changed to match the first one, but none of the other physical appearance. I repeated the experiment using only the UI (just eye color change) and just the UE (only name change) and also tried giving a human UI+UE to another human, which did result in the second human looking identical to the first and gaining the name. Since this seems to work with humans but not with any other race, I believe this is a bug.

Also worth noting that I think this may be connected to the fact that lizard and other races use mutant colors (hexadecimal) rather than a single-number skin tone like humans, and the fact that the frills didn't change might be attributed to the same issue that makes it so the wizard's magic mirror can't change the lizardman frill/horn style. This could explain why eye color is the only thing that changes with the UI injection, since that is shared between almost all races (except moths and plasmamen).

## Round ID:
I tested this on a self-hosted private server running Beestation's code.

## Testmerges:
N/A

## Reproduction:
Put a lizardman into the genetics machine, copy UI+UE and print an injector, then take a different lizardman and use the injector on them. For a control, do the same thing with two humans instead of lizards.

### PR DESCRIPTION
**Pull request by PowerfulBacon**
This was meant to be a quick fix, it took an hour and a half.

## About The Pull Request

fixes #210
(There was a reason this was the oldest still active issue, this one was much more of a pain in the ass than it first seems)

This adds a species mutant colour into the UI DNA block of a species. This means that injecting the UI of someone else into yourself will cause you adopt the other person's species colour.
Also renames the typepath for the scanner console since it was badly named and difficult to find on the item search.
~~Standardises mcolor to always use shorthex rather than 6 digit hex.~~ This was the case, however due to the preference freeze there is still an instance of it using 6 digit hex. Not that it matters anyway, it can support either.

## Why It's Good For The Game

Fixes lizards being unable to change their colour through genetics. There is a lot of other stuff that could be added, however it would end up bloating the UI code so moth wings will have to remain not attached to DNA for now (which is fine).

## Testing Photographs and Procedure

This was a bit of a pain in the ass trying to get the bodyparts to update if the mcolor was changed, since mcolor was considered to be something that doesn't change, however after a bunch of tests and compiles it works.
In the end this took a pretty big chunk of refactoring to get working.

image

## Changelog
:cl:
fix: Species mutant colour is now a part of the UI DNA, allowing you to change a lizards colour with the genetic console.
code: Renames the geneticist's DNA scanning console's typepath to scanconsole instead of scan_consolenew
/:cl:

### CODE REVIEW

**Comment by PowerfulBacon** [comment]
Needs mapping changes

**Comment by PowerfulBacon** [comment]
If someone reviews this I'll update the maps when its ready for merge.

**Comment by PowerfulBacon** [comment]
This is unlikely to get a review any time soon, so its pointless keeping this open since I'll just have to resolve merge conflicts every few days

**Review by itsmeow — COMMENTED** [review]
Not entirely a fan of the simplified hex, but I don't see any glaring issues here. If the PR is updated, it should be merged.

**Comment by EvilDragonfiend** [comment]
> Not entirely a fan of the simplified hex

Same here

**Comment by itsmeow** [comment]
@PowerfulBacon revive please

**Comment by PowerfulBacon** [comment]
> Not entirely a fan of the simplified hex, but I don't see any glaring issues here. If the PR is updated, it should be merged.

Everything, including hair, gets converted into simplified hex anyway when it passes through sanitize_hexcolor.

**Comment by Pirill** [comment]
Does this include Ethereal colours or do those use a special snowflake method?

**Comment by PowerfulBacon** [comment]
I am going to update them using a snowflake method that picks the nearest eth colour to the mutant colour possible

**Comment by PowerfulBacon** [comment]
Changing ethereal into a pink lizard
image
image

Changing a lizard into a blue-ish ethereal
image
image

**Comment by itsmeow** [comment]
> > Not entirely a fan of the simplified hex, but I don't see any glaring issues here. If the PR is updated, it should be merged.
> 
> Everything, including hair, gets converted into simplified hex anyway when it passes through sanitize_hexcolor.

Unsimplify it, we need to move away from 3 byte color to 6 byte like TG has, it's cleaner and more consistent. Having two color formats is really bad for the codebase

**Comment by PowerfulBacon** [comment]
That is a pretty big rework and out of scope for this PR. This PR fixes the attached issue and doesn't change anything related to the current storing of colours. If you want, I can revert the changing of things to 3 length colours so that in future an update can use them, however they will continue to be simplified down to 3 length colours.

File: _maps/_basemap.dm [hunk 1]
```diff
@@ -1,4 +1,4 @@
-//#define LOWMEMORYMODE //uncomment this to load centcom and runtime station and thats it.
+#define LOWMEMORYMODE //uncomment this to load centcom and runtime station and thats it.
```
Comment (PowerfulBacon) [inline _maps/_basemap.dm]:
```suggestion
//#define LOWMEMORYMODE //uncomment this to load centcom and runtime station and thats it.
```

**Comment by Tsar-Salat** [comment]
Just confirming, Transforming from one species to another (ex: Mutation Toxin) changes or doesnt change your mutant color?

**Comment by PowerfulBacon** [comment]
Not unless randomisation is involved. Ethereal colour and mutant colour will be linked with this PR, so if your mutant colour is red then your ethereal colour will be the closest matching red that is valid to ethereals.

**Comment by itsmeow** [comment]
> That is a pretty big rework and out of scope for this PR. This PR fixes the attached issue and doesn't change anything related to the current storing of colours. If you want, I can revert the changing of things to 3 length colours so that in future an update can use them, however they will continue to be simplified down to 3 length colours.

Yes, that's what I meant.

**Pull request closed without merging**